language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java

```diff
@@ -186,7 +186,7 @@ public static String getFillAllRecordFieldInsertText(Map<String, RecordFieldSymb
     public static Map<String, RecordFieldSymbol> getRecordFields(RawTypeSymbolWrapper<RecordTypeSymbol> wrapper,
                                                                  List<String> existingFields) {
         return wrapper.getRawType().fieldDescriptors().entrySet().stream()
-                .filter(e -> !existingFields.contains(e.getKey()))
+                .filter(e -> !existingFields.contains(e.getValue().getName().get()))
```

Comment:
Can't we use the key itself anyway it's `origField.name.value`?

Reply:
No when we are adding the key to the map we unescaped it and add them. [link](https://github.com/ballerina-platform/ballerina-lang/blob/446f250c44fa2adfd45193986da41a459b119459/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/util/FieldMap.java#L57)